python/ray/dashboard/client/src/pages/dashboard/memory/MemoryRowGroup.tsx

```diff
@@ -0,0 +1,148 @@
+import {
+  createStyles,
+  TableCell,
+  TableRow,
+  Theme,
+  withStyles,
+  WithStyles,
+} from "@material-ui/core";
+import AddIcon from "@material-ui/icons/Add";
+import RemoveIcon from "@material-ui/icons/Remove";
+import classNames from "classnames";
+import React from "react";
+import {
+  MemoryTableResponse,
+  MemoryTableEntry,
+  MemoryTableSummary,
+} from "../../../api";
+import MemorySummary from "./MemorySummary";
+
+const styles = (theme: Theme) =>
+  createStyles({
+    cell: {
+      padding: theme.spacing(1),
+      textAlign: "center",
+      "&:last-child": {
+        paddingRight: theme.spacing(1),
+      },
+    },
+    expandCollapseCell: {
+      cursor: "pointer",
+    },
+    expandCollapseIcon: {
+      color: theme.palette.text.secondary,
+      fontSize: "1.5em",
+      verticalAlign: "middle",
+    },
+    extraInfo: {
+      fontFamily: "SFMono-Regular,Consolas,Liberation Mono,Menlo,monospace",
+      whiteSpace: "pre",
+    },
+  });
+
+type Props = {
+  groupKey: string;
+  memoryTableGroups: MemoryTableResponse["group"];
+  initialExpanded: boolean;
+};
+
+type State = {
+  expanded: boolean;
+};
+
+class MemoryRowGroup extends React.Component<
+  Props & WithStyles<typeof styles>,
+  State
+> {
+  state: State = {
+    expanded: this.props.initialExpanded,
+  };
+
+  toggleExpand = () => {
```

Comment:
I recommend switching to a `useState` hook here, e.g. `expanded, setExpanded = useState(this.props.initialExpanded);`

Reply:
`toggleExpanded = () => setExpanded(!expanded)`